added devonfw/journey to build-and-deploy.yml

vlad961 opened the pull request:
updated build-and-deploy.yml with devonfw/journeys scripts, npm install and ng build.

vlad961 committed:
Update build-and-deploy.yml

Added devonfw/journeys scripts and build steps.

vlad961 committed:
Update build-and-deploy.yml

refactored the inserted sctructure of commands

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/devonfw/devonfw.github.io?pullRequest=324) <br/>All committers have signed the CLA.

vlad961 committed:
Merge branch 'devonfw:develop' into develop

SchettlerKoehler commented on .github/workflows/build-and-deploy.yml:
```diff
@@ -72,6 +76,18 @@ jobs:
           npm install
           cd ../../
           
+      - name: install for Journeys/app
```
"npm install for Journeys/app"

SchettlerKoehler commented on .github/workflows/build-and-deploy.yml:
```diff
@@ -72,6 +76,18 @@ jobs:
+      - name: npm install for Journey/scripts
+        run: |
+          cd journeys/scripts
+          npm install
```
I think this could be to late here as you are already calling the script in line 65

vlad961 replied:
I changed the order of the commands and put the npm install before the run command for the script. 
~~In case all jobs are done parallel (default behaviour if I understand the documentation correctly https://docs.github.com/en/actions/using-workflows/about-workflows#creating-dependent-jobs) shall I modify the run of the script like following:~~

~~name: Copy pom files for journeys~~
~~needs: npm install for Journeys/scripts~~
~~run: | node journeys/scripts/copyPomFiles.js~~

SchettlerKoehler commented on .github/workflows/build-and-deploy.yml:
```diff
@@ -154,6 +170,16 @@ jobs:
+      - name: Create output for journeys
+        run: |
+          chmod +x journeys/scripts/createOutput.sh
+          journeys/scripts/createOutput.sh
```
Please align with the rest of the team on the output path. Currently it looks strange

SchettlerKoehler commented on .github/workflows/build-and-deploy.yml:
```diff
@@ -154,6 +170,16 @@ jobs:
+      - name: Copy output of journeys
+        run: |
+          cp -avr journeys/target/generated-docs/ target/generated-docs/website/pages/journeys/
+          mv target/generated-docs/website/pages/journeys/journeys target/generated-docs/website/pages/journeys/journey
```
Why is that needed? It looks like createOutput is doing something in that direction already.

vlad961 committed:
Update build-and-deploy.yml

Name of npm install step refactored.
Changed order of build-and-deploy steps. Npm install first, afterwards the scripts are run. 
Adjusted the Create output for journeys step to save the json outputfiles in the journes/target/ directory.
The Copy output of journeys step is therefore needed now.

vlad961 committed:
Merge branch 'devonfw:develop' into develop

vlad961 committed:
Merge branch 'devonfw:develop' into develop

vlad961 committed:
Merge branch 'devonfw:develop' into develop

vlad961 committed:
Merge branch 'devonfw:develop' into develop

vlad961 committed:
Merge branch 'devonfw:develop' into develop

vlad961 committed:
Refacotring of build-and-deploy.yml

Delete unnecessary mv cmd.

vlad961 committed:
Merge branch 'devonfw:develop' into develop

vlad961 reviewed (commented):
commit approved
